docs/site/submitting_a_pr.md

```diff
@@ -0,0 +1,629 @@
+
+To flush out any **formatting** issues in code, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+To flush out any **syntax** issues in code, run `npm run build`. If there are issues, fix them, and run `npm run build` again. Repeat until all issues are gone.
+
+To flush out any **logical/execution** issues in code, run `npm run test`. This will run the existing mocha test cases, and any new test cases you have added. If there are issues, fix them, run `npm build`, and run `npm run test` again. Repeat until all issues are gone.
+
+### 11. Verifying your documentation changes
+
+To flush out any **formatting** issues in documentation, run `npm run lint`. If there are issues, run `npm run lint:fix`.
+
+Documentation written in **Markdown** format needs to be compiled into **HTML** files. To flush out any layout or substitution issues ( from using 'includes' for example), run `npm run build:site`. If there are any issues, fix them and run `npm run build:site` again.
+Then `cd sandbox/loopback.io/` and run `npm start` to start a local web server that hosts the HTML documentation.
+
+![submit_pr_check_local_docs_1.png](./imgs/submit_pr_check_local_docs_1.png)
+
+
+In your browser, navigate to the specified server address, click on `Docs`, click on `Loopback 4`, and verify that your document changes appear and are well-rendered.
+
+Press `Ctrl+C` to stop the local documentation server.
+
+
+
+
+### 12. Staging your changes
+
+When you are pleased with the work you have done on the local copy of your feature branch, you will want to `stage` your changes in preparation for a `commit`.
+
+To find out what has changed in your local copy of your feature branch, type :
+
+```
+git status
+```
+
+In my case I have :
+
+![submit_pr_staging_1.png](./imgs/submit_pr_staging_1.png)
+
+There is a new file `docs/site/submitting_a_pr.md` that is `untracked` and needs to be `staged`.
+
+To stage all untracked files, you can type:
+
+```
+git add --all
+```
+
+To stage one untracked file at a time, you can type:
+
+```
+git add {relative path to file from root directory}
+```
+
+In my case, this would be
+
+```
+git add docs/site/submitting_a_pr.md
+```
+
+To see which files are `staged` and `unstaged`, type:
+
+```
+git status
+```
+![submit_pr_staging_2.png](./imgs/submit_pr_staging_2.png)
+
+In my case, the file `docs/site/submitting_a_pr.md` is properly `staged`, and there are no `unstaged (untracked)` files.
+
+### 13. Committing your changes
+
+When you are pleased with your staged changes, it is time to create a `commit` and give it a message.
+
+Please read the [Commit Message Format](https://loopback.io/doc/en/contrib/code-contrib-lb4.html#commit-message-guidelines) guidelines to correctly format your commit message.
+
+To help with abiding by the rules of commit messages, please use the `commitizen` tool mentioned in the documentation above. This means will we use `git cz` instead of `git commit`.
```

Comment:
Does `git cz` support add sign-off message? After switching to DCO every commit needs author to sign off.
Would be equivalent to `git commit -s -m 'fix: some description'`

Reply:
`commitizen` does support signoff.  when you enter the long description, you specify:

```
my long description.\n\nSigned-off-by: Dominique Emond <dremond@ca.ibm.com>
```

and this would provide a commit message like:

```
docs: tutorial on how to submit a pull request to lb4
    
This tutorial shows contributors step-by-step instructions on how to submit a pull request (PR) to Loopback v4  

Signed-off-by: Dominique Emond <dremond@ca.ibm.com>
```

I initially prepared my tutorial for steps involving DCO, but @dhmlau has informed me to prepare it with the `CLA` approach; since DCO is not approach is not 100% certain.